test_launch_ros/test/test_launch_ros/actions/test_lifecycle_node.py

```diff
@@ -51,5 +51,5 @@ def test_node_name():
         namespace='my_ns',
     )
     lc = LaunchContext()
-    node_object._perform_substitutions(lc)
+    node_object._Node__node_desc._perform_substitutions(lc, [])
```

Comment:
@roger-strain is it necessary to access a "private" attribute like this? Isn't the functionality under test observable via public API? Same everywhere else below.

Reply:
It actually wasn't (easily) available without making some assumptions about data types that I didn't want to make. I've added a new getter on the `actions.Node` for `ros_exec`, which will allow us to easily get access to the ROS-style executable, rather than the more generic `Executable` description, which didn't provide the things we needed.